src/Microsoft.IdentityModel.Logging/IdentityModelEventSource.cs

```diff
@@ -61,7 +61,38 @@ private IdentityModelEventSource()
         /// <summary>
         /// String that is used in place of any arguments to log messages if the 'ShowPII' flag is set to false.
         /// </summary>
-        public static string HiddenPIIString { get; } = "[PII is hidden by default. Set the 'ShowPII' flag in IdentityModelEventSource.cs to true to reveal it.]";
+        public static string HiddenPIIString { get; } = "[PII is hidden]";
+
+        /// <summary>
+        /// Indicates whether or the log message header (contains library version, date/time, and PII debugging information) has been written.
+        /// </summary>
+        public static bool HeaderWritten { get; set; } = false;
+
+        /// <summary>
+        /// The current library version.
+        /// </summary>
+        private static string _version = "5.2.2";
```

Comment:
Use a dynamic technique to obtain the version.
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/master/src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectMessage.cs#L196

Reply:
Makes sense. Just made the change.